src/goto-symex/symex_function_call.cpp

```diff
@@ -431,28 +427,20 @@ void goto_symext::return_assignment(statet &state)
 
   target.location(state.guard.as_expr(), state.source);
 
-  if(code.operands().size()==1)
-  {
-    exprt value=code.op0();
+  PRECONDITION(code.operands().size() == 1);
+  
+  exprt value=code.op0();
 
-    if(frame.return_value.is_not_nil())
-    {
-      code_assignt assignment(frame.return_value, value);
+  if(frame.return_value.is_not_nil())
+  {
+    code_assignt assignment(frame.return_value, value);
 
-      if(!base_type_eq(assignment.lhs().type(),
-                       assignment.rhs().type(), ns))
-        throw
-          "goto_symext::return_assignment type mismatch at "+
-          instruction.source_location.as_string()+":\n"+
```

Comment:
Here I made the judgement call that the error message, especially within the context, was too chatty. I replaced it with an `INVARIANT` which keeps part of the message, but drops the extraneous information (at least for the usual case)

Reply:
or to put in in another way, I have a severe dislike for an error message in an invariant that prints two `expressions.pretty()`

Reply:
a) - Are we sure this is not a user level error?
b) Imagine you made a modification to CBMC and hit this new INVARIANT - how would you go about debugging it? What types mismatch?